n2y/config.py

```diff
@@ -1,41 +1,134 @@
+import yaml
 
 from n2y.utils import strip_hyphens
 
 
 logger = logging.getLogger(__name__)
 
 
-def database_config_json_to_dict(config_json):
+DEFAULTS = {
+    "media_root": "media",
+    "media_url": "./media/",
+}
+
+
+EXPORT_DEFAULTS = {
+    "id_property": None,
+    "content_property": None,
+    "url_property": None,
+    "notion_filter": [],
+    "notion_sorts": [],
+    "pandoc_format": "gfm+tex_math_dollars+raw_attribute",
+    "pandoc_options": [
+        '--wrap', 'none',  # don't hard line-wrap
+        '--eol', 'lf',  # use linux-style line endings
+    ],
+    "plugins": [],
+    "property_map": {},
+}
+
+
+def load_config(path):
     try:
-        config = json.loads(config_json)
-    except json.JSONDecodeError as exc:
-        logger.error("Error parsing the data config JSON: %s", exc.msg)
+        with open(path, "r") as config_file:
+            config = yaml.safe_load(config_file)
+    except yaml.YAMLError as exc:
+        logger.error("Error parsing the config file: %s", exc)
+        return None
+    except FileNotFoundError:
+        logger.error("The config file '%s' does not exist", path)
         return None
-    if not validate_database_config(config):
+    if not validate_config(config):
+        logger.error("Invalid config file: %s", path)
         return None
+
+    defaults_copy = copy.deepcopy(DEFAULTS)
+    config = {**defaults_copy, **config}
+
+    merged_exports = merge_config(
+        config.get("exports", []),
+        EXPORT_DEFAULTS,
+        config.get("export_defaults", {}),
+    )
+    config["exports"] = merged_exports
     return config
 
 
-def validate_database_config(config):
-    try:
-        for database_id, config_values in config.items():
-            if not _valid_id(database_id):
-                logger.error("Invalid database id in database config: %s", database_id)
-                return False
-            for key, values in config_values.items():
-                if key not in ["sorts", "filter"]:
-                    logger.error("Invalid key in database config: %s", key)
-                    return False
-                if not isinstance(values, dict) and not isinstance(values, list):
-                    logger.error(
-                        "Invalid value of type '%s' for key '%s' in database config, "
-                        "expected dict or list", type(values), key,
-                    )
-                    return False
-    except AttributeError:
+def merge_config(config_items, builtin_defaults, defaults):
+    """
+    For each config item, merge in both the user provided defaults and the
+    builtin defaults for each key value pair."
+    """
+    merged_config_items = []
+    for config_item in config_items:
+        master_defaults_copy = copy.deepcopy(builtin_defaults)
+        defaults_copy = copy.deepcopy(defaults)
+        config_item_copy = copy.deepcopy(config_item)
+        merged_config_item = {**master_defaults_copy, **defaults_copy, **config_item_copy}
```

Comment:
Didn't know you could update dictionaries this way. Cool!

Reply:
I'm sure you looked into this but I assume that with this syntax, the dictionaries get updated in the order they appear right? So it's equivalent to:

```
merged_config_item = master_defaults_copy
merged_config_item = merged_config_item.update(defaults_copy)
merged_config_item = merged_config_item.update(config_item_copy)
```